cmd/deploy.go

```diff
@@ -13,24 +14,58 @@ type DeployCmd struct {
 	ui              boshui.UI
 	deployment      boshdir.Deployment
 	releaseUploader ReleaseUploader
+	director        boshdir.Director
 }
 
 type ReleaseUploader interface {
 	UploadReleases([]byte) ([]byte, error)
 	UploadReleasesWithFix([]byte) ([]byte, error)
 }
 
+type Conf struct {
+	Flags              []string `yaml:"flags"`
+	IncludeDeployments []string `yaml:"include"`
+	ExcludeDeployments []string `yaml:"exclude"`
+}
+
 func NewDeployCmd(
 	ui boshui.UI,
 	deployment boshdir.Deployment,
 	releaseUploader ReleaseUploader,
+	director boshdir.Director,
 ) DeployCmd {
-	return DeployCmd{ui, deployment, releaseUploader}
+	return DeployCmd{ui, deployment, releaseUploader, director}
 }
 
 func (c DeployCmd) Run(opts DeployOpts) error {
 	tpl := boshtpl.NewTemplate(opts.Args.Manifest.Bytes)
 
+	configs, _ := c.director.ListConfigs(1, boshdir.ConfigsFilter{Type: "deploy"})
+
+	for _, config := range configs {
+		var conf Conf
+
+		yaml.Unmarshal([]byte(config.Content), &conf)
+
+		deploymentIncluded := containsDeployment(conf.IncludeDeployments, c.deployment.Name())
+		deploymentExcluded := containsDeployment(conf.ExcludeDeployments, c.deployment.Name())
+
+		if conf.ExcludeDeployments != nil &&
+			conf.IncludeDeployments != nil {
+			c.ui.PrintLinef("Ignoring deployment flags from config of type '%s' (name: '%s'). Please use only 'include'- OR 'exclude'-property in the config.", config.Type, config.Name)
+		} else {
+			if (conf.IncludeDeployments == nil && conf.ExcludeDeployments == nil) ||
+				deploymentIncluded ||
+				(!deploymentExcluded && conf.ExcludeDeployments != nil) {
```

Comment:
I first thought of such pseudo-code instead:
```
decision=NO
IF includeList == null OR includeList contains deploymentName THEN decision=YES
IF excludeList != null AND excludeList contains deploymentName THEN decision=NO
```

Then, inspired by the [director’s code for addons](https://github.com/cloudfoundry/bosh/blob/main/src/bosh-director/lib/bosh/director/addon/addon.rb#L39-L42), here is a better version (see other commend below), which is equivalent:
```
decision = applies(includeList, deploymentName) && !applies(excludeList, deploymentName)
```
This is easier to understand, and consistent with what we already have for other include/exclude mechanisms in Bosh.

Reply:
@bgandon thanks for the review. I agree that would be easier to understand, however only having the check 
`decision = applies(includeList, deploymentName) && !applies(excludeList, deploymentName)`

would not apply to the following cases, that are currently taking care of:
1. No warning/ignoring of the flags in case both include and exclude properties are maintained.
2. If the exclude property is empty, it will always be a NO decision (considering that the containsDeployment() -> applies() method gets rewritten to return true if `list == null OR list contains deploymentName`)

So I am not sure about that...

Reply:
That’s right, my second suggestion is not equivalent to the first one, and i didn’t take into account the ambiguous case where both include and exclude are specified.

The idea with my first suggestion, was to skip the ambiguity and just le the exclude win when both are contradictory concerning a given deployment name.

I think we can go ahead with the code as is.